plugins/modules/ec2_vpc_peer.py

```diff
@@ -335,237 +337,259 @@
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import boto3_tag_specifications
 from ansible_collections.amazon.aws.plugins.module_utils.transformation import ansible_dict_to_boto3_filter_list
 
 from ansible_collections.community.aws.plugins.module_utils.modules import AnsibleCommunityAWSModule as AnsibleAWSModule
 
 
-def wait_for_state(client, module, state, pcx_id):
+def wait_for_state(client, module: AnsibleAWSModule, state: str, peering_id: str) -> NoReturn:
     waiter = client.get_waiter("vpc_peering_connection_exists")
-    peer_filter = {
-        "vpc-peering-connection-id": pcx_id,
+    filters = {
+        "vpc-peering-connection-id": peering_id,
         "status-code": state,
     }
     try:
-        waiter.wait(Filters=ansible_dict_to_boto3_filter_list(peer_filter))
+        waiter.wait(Filters=ansible_dict_to_boto3_filter_list(filters))
     except botocore.exceptions.WaiterError as e:
         module.fail_json_aws(e, "Failed to wait for state change")
     except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
         module.fail_json_aws(e, "Enable to describe Peerig Connection while waiting for state to change")
 
 
-def describe_peering_connections(params, client):
-    peer_filter = {
+def describe_peering_connections(client, module: AnsibleAWSModule, params) -> Dict[str, Any]:
+    peering_connections: Dict = {}
+
+    filters = {
         "requester-vpc-info.vpc-id": params["VpcId"],
         "accepter-vpc-info.vpc-id": params["PeerVpcId"],
     }
-    result = client.describe_vpc_peering_connections(
-        aws_retry=True,
-        Filters=ansible_dict_to_boto3_filter_list(peer_filter),
-    )
-    if result["VpcPeeringConnections"] == []:
-        # Try again with the VPC/Peer relationship reversed
-        peer_filter = {
-            "requester-vpc-info.vpc-id": params["PeerVpcId"],
-            "accepter-vpc-info.vpc-id": params["VpcId"],
-        }
-        result = client.describe_vpc_peering_connections(
-            aws_retry=True,
-            Filters=ansible_dict_to_boto3_filter_list(peer_filter),
+
+    try:
+        peering_connections = describe_vpc_peering_connections(
+            client, **{"Filters": ansible_dict_to_boto3_filter_list(filters)}
         )
+        if peering_connections["VpcPeeringConnections"] == []:
+            # Try again with the VPC/Peer relationship reversed
+            filters = {
+                "requester-vpc-info.vpc-id": params["PeerVpcId"],
+                "accepter-vpc-info.vpc-id": params["VpcId"],
+            }
+            peering_connections = describe_vpc_peering_connections(
+                client, **{"Filters": ansible_dict_to_boto3_filter_list(filters)}
+            )
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
-    return result
+    return peering_connections["VpcPeeringConnections"]
```

Comment:
```suggestion
    return peering_connections
```
can be updated according to https://github.com/ansible-collections/amazon.aws/pull/2303/files#r1778361791